### ISSUE TITLE
Private Service Access support

### ISSUE THREAD

**Issue by maximumG** [terraform-google-modules/terraform-google-network#510]
### TL;DR

the `[terraform-google-network](https://github.com/terraform-google-modules/terraform-google-network)` and its submodule seems to not provide any out of the box support for configuring a Private Service Access.

We would be eager to have a submodule handling such architecture to automatically provision PSA ranges and a peering with google-managed VPC networks for AlloyDB and MemoryStore.

### Additional information

Google Cloud Fabric modules handles the PSA configuration out of the box [here](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/net-vpc/psa.tf)

**Comment by imrannayer** [comment]
Private service access module is a [sub-module of cloudSql module](https://github.com/terraform-google-modules/terraform-google-sql-db/tree/master/modules/private_service_access).

https://github.com/terraform-google-modules/terraform-google-sql-db/tree/master/modules/private_service_access

**Comment by maximumG** [comment]
Many thanks for pointing this out, I didn't check the cloudSql module.

I would advocate to move this module in the _terraform-google-network_ repository as it is not only related to cloudSQL. Indeed, we need such PSA if we want to spawn AlloyDB or MemoryStore as well and get private connectivity within our VPC. Furthermore, in a shared VPC context, team owning the databases resources might not have the right to deal with the shared VPC configuration. I really think that this peering should be handle at the network team level, to avoid any DB engine team to ever bother with the underlying VPC infrastructure when spawning a DB.

WDYT ?

**Comment by imrannayer** [comment]
Logically it make sense but as of now a lot of customers are using it. Just because it is a sub-module of database module does not mean DBAs have to do it. Network team can use this sub-module. DIfferent teams use different sub-modules all the time depending on the use case.

**Comment by maximumG** [comment]
make sense :smile: . Thanks for your answer. As you mentioned we will use this PSA module in our own GCP terraform network blueprint
